Tus uploader `onShouldRetry` has incorrect type signature

### Initial checklist

- [X] I understand this is a bug report and questions should be posted in the [Community Forum](https://community.transloadit.com/)
- [X] I searched [issues](https://github.com/transloadit/uppy/issues?q=is%3Aissue) and couldn’t find anything (or linked relevant results below)

### Link to runnable example

_No response_

### Steps to reproduce

Use `onShouldRetry` as per docs in https://github.com/transloadit/uppy/blob/uppy%404.1.0/docs/uploader/tus.mdx#onshouldretry-err-retryattempt-options-next. Can just copy that code.

### Expected behavior

Type check (`tsc`) completes successfully.

### Actual behavior

```
error TS2322: Type '(err: DetailedError, _retryAttempt: number, _options: TusOpts<Meta, Body>, next: (e: DetailedError) => void) => true | void' is not assignable to type '(err: DetailedError, retryAttempt: number, options: TusOpts<Meta, Body>, next: (e: DetailedError) => void) => boolean'.
  Type 'true | void' is not assignable to type 'boolean'.
    Type 'void' is not assignable to type 'boolean'.

122             onShouldRetry: (err, _retryAttempt, _options, next) => {
                ~~~~~~~~~~~~~

  node_modules/@uppy/tus/lib/index.d.ts:22:5
    22     onShouldRetry?: (err: tus.DetailedError, retryAttempt: number, options: TusOpts<M, B>, next: (e: tus.DetailedError) => void) => boolean;
           ~~~~~~~~~~~~~
    The expected type comes from property 'onShouldRetry' which is declared here on type 'TusOpts<Meta, Body>'
```

This is due to the fact that the type of the `next` function passed in returns `void` instead of `boolean`, which is required for the example code in the documentation to work.

--- a/packages/@uppy/tus/src/index.ts
+++ b/packages/@uppy/tus/src/index.ts
@@ -53,7 +53,7 @@ export interface TusOpts<M extends Meta, B extends Body>
     err: tus.DetailedError,
     retryAttempt: number,
     options: TusOpts<M, B>,
-    next: (e: tus.DetailedError) => void,
+    next: (e: tus.DetailedError) => boolean,
   ) => boolean
   retryDelays?: number[]
   withCredentials?: boolean

